### DISCUSSION TITLE
useCopyToClipboard reset `value`

### DISCUSSION THREAD

**Discussion by benjick** [juliencrn/usehooks-ts#641] [Help]
Hello

I'm using `useCopyToClipboard` in a list, and I have some code like this:

```tsx
<Button
  variant="secondary"
  className="shrink-0"
  size="icon"
  onClick={handleCopy}
>
  {copyValue === url ? (
    <CheckIcon className="h-4 w-4 text-green-500" />
  ) : (
    <ClipboardCopyIcon className="h-4 w-4" />
  )}
</Button>
```

When I copy the first row; the CheckIcon renders. Then when I copy the second row the `value` isn't updated in the first one. I'm guessing this is because observing the clipboard is impossible. Are there any recommended workarounds for this? Do I need to implement a context that keeps track of the last copied value?

Cheers

**Comment by tamish-max** [comment]
Your guess is right: `useCopyToClipboard` only remembers what **that hook instance** last copied. It does **not** observe the system clipboard, so the first row has no way to know that another row copied something later.

If you want a single “currently copied row” indicator, I’d lift that state up to the parent (or a context), e.g.:

```tsx
const [lastCopied, setLastCopied] = useState<string | null>(null)
const [, copy] = useCopyToClipboard()

async function handleCopy(url: string) {
  const ok = await copy(url)
  if (!ok) return

  setLastCopied(url)
  setTimeout(() => setLastCopied(null), 1500)
}
```

Then each row renders from `lastCopied === url`.

That gives you deterministic UI state without depending on clipboard observation.
